Document terms types addition process (#82)
OpenTermsArchive/terms-types: CONTRIBUTING.md

If you need to track some terms for a service, you might encounter cases when no matching type seems to be available in the list of supported terms types.

### Open a discussion

If no existing type matches the terms you intend to track, check the open discussions in the category “New Type” in the [Discussions section](https://github.com/opentermsarchive/terms-types/discussions) of this repository.

If no discussion seems to match the type you are interested in adding, [start a new “New Type” discussion](https://github.com/OpenTermsArchive/terms-types/discussions/new?category=new-type) and call it “Add _your proposed terms type name_”.

### Define the type name

Propose a unique and descriptive name for the new terms type.

It should be the most commonly used and most internationally understandable for this type. It doesn't have to, and probably should not, match the service-provided name. Make use of tools such as Google Trends, Linguee and thesaurus to find the name that strikes the best balance between generality, precision and multilingualism.

> > For example, the “Imprint” type is [called this way](https://github.com/OpenTermsArchive/engine/issues/801) as a balance between the French “mentions légales”, the German “impressum”, and other considerations.

Author: I've added this information to the discussion template, and you can see what it looks like on [my fork](https://github.com/clementbiron/terms-types/discussions/new?category=suggestions).

> > For example, the “Imprint” type is [called this way](https://github.com/OpenTermsArchive/engine/issues/801) as a balance between the French “mentions légales”, the German “impressum”, and other considerations.

Reviewer: Where is this added to the template? I don't see the change in the PR :)

Ensure that this name (or any close one) is not already used in the [database](./termsTypes.json).

> Note that service-specific types (such as “Twitter Privacy Policy”) are not allowed. Terms types aim at allowing comparisons across services and should thus be generic.

Author: Yes but

> Ensure that this name (or any close one) is not already used in the [database](./termsTypes.json).

Seems to me redondant to `First of all, double-check that it does not already exist in the [database](./termsTypes.json). If it does, use the associated type for tracking.` line 24.

I have reintroduced the notice on the service-specific types.

> Then check the [open discussions](https://github.com/opentermsarchive/terms-types/discussions) for this repository to make sure that there is not already a discussion about the type you intend to add.

Reviewer: ```suggestion
Then check the [open discussions](https://github.com/opentermsarchive/terms-types/discussions) in this repository to make sure that there is not already a discussion about the type you intend to add.
```

> If no discussion seems to match the type you are interested in adding, [start a new discussion](https://github.com/OpenTermsArchive/terms-types/discussions/new?category=submitted) by filling in this form.

Reviewer: “this form” is ambiguous since it doesn't have a link 🙂 

```suggestion
If no discussion seems to match the type you are interested in adding, [start a new discussion](https://github.com/OpenTermsArchive/terms-types/discussions/new?category=submitted).
```

> The Core Team reviews the term type additions categorized as `Submitted` to ensure that all the necessary information is gathered and that at least 5 examples are provided by maximising industry and jurisdiction diversity. When this is the case, the discussion is categorized as `Qualified` and opened up to the community.

Reviewer: ```suggestion
New suggestions for terms types are categorized as `Submitted`. The Core Team reviews these suggestions and ensures that all the necessary information is gathered and clear and that at least 5 examples are provided, maximizing industry and jurisdiction diversity. When this is the case, the discussion is moved to the `Qualified` category and opened up to the community.
```

> Build consensus by publicizing your suggestion across Open Terms Archive channels and engaging the community. Throughout the discussion, the data should be critically examined, considering international perspectives on phrasing and the variety of definitions across jurisdictions.

Reviewer: ```suggestion
Once your suggestion is `Qualified`, build consensus by publicizing it across Open Terms Archive channels and engaging with the community. Throughout the discussion, the data should be critically examined, especially taking into account international perspectives on vocabulary and the variety of definitions across jurisdictions.
```

> Once a consensus has been reached and at least 2 weeks have passed since the discussion was opened to ensure visibility within the community, a pull request for the addition to the database must be submitted. To facilitate tracking and participation, each pull request will contain only one type addition and will link to the corresponding discussion.

Reviewer: ```suggestion
Once a consensus has been reached and at least 2 weeks have passed since the discussion was opened, to ensure visibility within the community, a pull request for the addition to the database can be submitted. To facilitate tracking and participation, each pull request should contain only one type addition and must link to the corresponding discussion.
```

> The Core Team will then review the pull request, taking into account factors such as uniqueness, clarity, and relevance.

Reviewer: I feel that the final review should only be about formatting, and that any concerns on uniqueness, clarity and relevance should have been addressed in the discussion 🤔

Author: yes!

First of all, thanks for taking the time to contribute! 🎉👍

## Table of Contents

- [Add new Terms Types](#add-new-terms-types)
  - [Open a discussion](#open-a-discussion)
  - [Integration process](#integration-process)
- [Governance considerations](#governance-considerations)
- [List a new contributor](#list-a-new-contributor)
- [Changelog](#changelog)
  - [Modified terms types or data structure](#modified-terms-types-or-data-structure)
  - [Changes that do not impact users](#changes-that-do-not-impact-users)

- - -

## Add new Terms Types

Every terms tracked in Open Terms Archive must have a type. Each service can have only one terms of a given type.

If you need to track some terms for a service, you might encounter cases when no matching type seems to be available in the list of supported terms types.

### Open a discussion

> First of all, double-check that it does not already exist in the [database](./termsTypes.json). If it does, use the associated type for tracking. 

Reviewer: ```suggestion
First of all, double-check that the type you consider adding does not already exist in the [database](./termsTypes.json). If it does, use the associated type for tracking. 
```

> Then check the [open discussions](https://github.com/opentermsarchive/terms-types/discussions) in this repository to make sure that there is not already a discussion about the type you intend to add.

Reviewer: ```suggestion
If not, then check the [open discussions](https://github.com/opentermsarchive/terms-types/discussions) in this repository to make sure that there is not already a discussion about the type you consider adding.
```

If no discussion seems to match the type you are interested in adding, [start a new discussion](https://github.com/OpenTermsArchive/terms-types/discussions/new?category=submitted).

Note that service-specific types (such as “Twitter Privacy Policy”) are not allowed. Terms types aim at allowing comparisons across services and should thus be generic.

### Integration process

> New suggestions for terms types are categorized as `Submitted`. The Core Team reviews these suggestions and ensures that all the necessary information is gathered and clear and that at least 5 examples are provided, maximizing industry and jurisdiction diversity. When this is the case, the discussion is moved to the `Qualified` category and opened up to the community.

Reviewer: ```suggestion
New suggestions for terms types are categorized as `Submitted`. The Core Team reviews these suggestions and ensures that all the necessary information is gathered and clear and that at least 5 examples are provided, maximizing industry and jurisdiction diversity. Once that is the case, the discussion is moved to the `Qualified` category and opened up to the community.
```

Once your suggestion is `Qualified`, build consensus by publicizing it across Open Terms Archive channels and engaging with the community. Throughout the discussion, the data should be critically examined, especially taking into account international perspectives on vocabulary and the variety of definitions across jurisdictions.

> Once a consensus has been reached and at least 2 weeks have passed since the discussion was opened, to ensure visibility within the community, a pull request for the addition to the database can be submitted. To facilitate tracking and participation, each pull request should contain only one type addition and must link to the corresponding discussion.

Reviewer: ```suggestion
Once a consensus has been reached and at least 2 weeks have passed since the discussion was opened (to ensure visibility within the community), a pull request for the addition to the database can be submitted. To facilitate tracking and participation, each pull request should contain only one type addition and must link to the corresponding discussion.
```
